src/main/java/com/microsoft/jfr/FlightRecorderConnection.java

```diff
@@ -113,19 +100,12 @@ public long startRecording(RecordingOptions recordingOptions, RecordingConfigura
             String[] argTypes = new String[]{};
             final long id = (long) mBeanServerConnection.invoke(objectName, "newRecording", args, argTypes);
 
-            if (recordingConfiguration != null) {
-                String configuration = recordingConfiguration.getConfiguration();
-                if (configuration != null && configuration.trim().length() > 0) {
-                    args = new Object[]{id, configuration};
-                    argTypes = new String[]{long.class.getName(), String.class.getName()};
-                    mBeanServerConnection.invoke(objectName, recordingConfiguration.getMbeanSetterFunction(), args, argTypes);
-                }
-            }
+            setOptions(recordingConfiguration, id);
 
             if (recordingOptions != null) {
-                Map<String,String> options = recordingOptions.getRecordingOptions();
+                Map<String, String> options = recordingOptions.getRecordingOptions();
```

Comment:
I'd like to see a `setOptions` method that parallels the `setConfiguration` method so that this code looks like
```
if (recordingConfiguration != null) {
    setConfiguration(recordingConfiguration, id);
}

if (recordingOptions != null) {
    setOptions(recordingOptions, id);
}

Reply:
Done (this type of refactoring might be outside this PR)

Reply:
Yes, I did consider that.